sureal/dataset_reader.py

```diff
@@ -521,6 +521,76 @@ def to_pc_dataset_file(self, dataset_filepath, **kwargs):
         pc_dataset = self.to_pc_dataset(**kwargs)
         self.write_out_dataset(pc_dataset, dataset_filepath)
 
+    def to_combined_overlap_dataset(self, second_dataset_reader, **kwargs):
+        """
+        A function to find an overlap between self.dataset and the second dataset and combine the scores for the
+        overlapping videos. The overlap is determined by matching paths.
+        """
+        assert type(second_dataset_reader).__name__ == 'RawDatasetReader', 'RawDatasetReader can only be combined ' \
+                                                                           'with another RawDatasetReader'
+
+        newone = self._prepare_new_dataset(kwargs)
+        newone.ref_videos = []
+        newone.dis_videos = []
+
+        content_ids_in = []
+        warning_printed = False
+        for vid1 in self.dataset.dis_videos:
```

Comment:
Thanks for the comments and tips. Two questions:

Re 2): does it make sense to create a dictionary from the lists? The current logic is - if both datasets are dictionary style, create a dictionary and if any of the subjects voted in both datasets, make it a repetition. If one of the datasets is a different style, make both of them a list and simply append the scores. Are you suggesting that we create a dictionary with generic names from the list/tuple style datasets?

Re 3): I'm afraid this only works when the `new_dis_videos['os']` is already a list, doesn't it. However, it can be a single value (float) or a list.

Reply:
Re 2): Yes, I am thinking dictionary with generic names as the "normalized" style. Thinking back, perhaps I should have used one style only, which is the dictionary style. This is more generic and compact than the list style. 

Re 3): would `'os': [90.0]` and `'os': 90.0` both be parsable by the code? I think so, right?

Reply:
2) OK, that makes sense. We can start moving towards mostly creating the dictionary style.

3) Yes they are both possible but the one-liner you suggested would throw an error in the latter case because it tries to append a float.

Reply:
Re 3): right, in this case, you will have to test on the instance type no matter what to differentiate list or float.